opencog/distvalue/DVecUtils.h

```diff
@@ -0,0 +1,89 @@
+/*
+ * opencog/distvalue/DVecUtils.h
+ *
+ * Copyright (C) 2018 SingularityNet
+ * All Rights Reserved
+ *
+ * Written by Roman Treutlein
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License v3 as
+ * published by the Free Software Foundation and including the exceptions
+ * at http://opencog.org/wiki/Licenses
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program; if not, write to:
+ * Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+
+#ifndef _OPENCOG_CTUTILS_H
+#define _OPENCOG_CTUTILS_H
+
+#include <vector>
+
+namespace opencog
+{
+
+/*
+ * Utility functions for working with std::vector<double> "DVec" and
+ * std::vector<std::vector<double>> "DVecSeq"
+ */
+
+typedef std::vector<double> DVec;
+typedef std::vector<DVec> DVecSeq;
+
+//Element wise Operators
+//returns vector of size min(a.size(),b.size())
+DVec operator+(const DVec & a, const DVec & b);
+DVec operator-(const DVec & a, const DVec & b);
+DVec operator*(const DVec & a, const DVec & b);
+DVec operator/(const DVec & a, const DVec & b);
```

Comment:
You might want to take a look at https://www.boost.org/doc/libs/1_71_0/libs/numeric/ublas/doc/index.html, not necessarily suggesting you use it, it might be overkill, just wanted to mention its existence.